lib/XTR/StatsArchiveTask.php

```diff
@@ -205,11 +214,22 @@ private function tidyStats()
             $maxAttempts = $this->getOption('statsDeleteMaxAttempts', 10);
             $statsDeleteSleep = $this->getOption('statsDeleteSleep', 3);
 
-            $options = [
-                'maxAttempts' => $maxAttempts,
-                'statsDeleteSleep' => $statsDeleteSleep,
-                'limit' => 10000
-            ];
+            if ($this->timeSeriesStore->getEngine() == 'mongodb') {
```

Comment:
we've got this handled here, but it seems like we should handle that inside the mongotimeseriesstore?

If we have a restriction in Mongo, then we ought to ignore the limit option internally i think, rather than handling it out here